core/model/src/payment.rs

```diff
@@ -69,37 +70,62 @@ pub mod local {
     }
 
     #[derive(Clone, Debug, Serialize, Deserialize)]
-    pub struct Init {
-        pub identity: NodeId,
-        pub provider: bool,
-        pub requestor: bool,
+    pub struct RegisterAccount {
+        pub driver: String,
+        pub platform: String,
+        pub address: String,
+        pub mode: AccountMode,
     }
 
-    impl RpcMessage for Init {
-        const ID: &'static str = "init";
+    #[derive(Clone, Debug, Serialize, Deserialize, thiserror::Error)]
+    pub enum RegisterAccountError {
+        #[error("Account already registered")]
+        AlreadyRegistered,
+        #[error("Error while registering account: {0}")]
+        Other(String),
+    }
+
+    impl RpcMessage for RegisterAccount {
+        const ID: &'static str = "RegisterAccount";
         type Item = ();
-        type Error = GenericError;
+        type Error = RegisterAccountError;
     }
 
     #[derive(Clone, Debug, Serialize, Deserialize)]
-    pub struct GetStatus(NodeId);
+    pub struct UnregisterAccount {
+        pub platform: String,
+        pub address: String,
+    }
 
-    impl From<NodeId> for GetStatus {
-        fn from(id: NodeId) -> Self {
-            GetStatus(id)
-        }
+    #[derive(Clone, Debug, Serialize, Deserialize, thiserror::Error)]
+    pub enum UnregisterAccountError {
+        #[error("Account not registered")]
+        NotRegistered,
+        #[error("Error while unregistering account: {0}")]
+        Other(String),
     }
 
-    impl AsRef<NodeId> for GetStatus {
-        fn as_ref(&self) -> &NodeId {
-            &self.0
-        }
+    impl RpcMessage for UnregisterAccount {
+        const ID: &'static str = "UnregisterAccount";
+        type Item = ();
+        type Error = UnregisterAccountError;
     }
 
-    impl GetStatus {
-        pub fn identity(&self) -> NodeId {
-            self.0
-        }
+    #[derive(Clone, Debug, Serialize, Deserialize)]
+    pub struct NotifyPayment {
+        pub driver: String,
+        pub platform: String,
+        pub amount: BigDecimal,
+        pub sender: String,
+        pub recipient: String,
+        pub order_ids: Vec<String>,
+        pub confirmation: PaymentConfirmation,
+    }
+
+    #[derive(Clone, Debug, Serialize, Deserialize)]
+    pub struct GetStatus{
+        pub platform: String,
```

Comment:
Same here.

Reply:
Here it can't be done away with because we don't know the driver. `address` is not enough to establish that. There can be two different drivers handling e.g. GNT and ETH transactions but using the same account address.